src/algebra/chinese-remainder-theorem.md

```diff
@@ -31,12 +31,59 @@ is equivalent to the system of equations
 
 $$\begin{align}
     x &\equiv a_1 \pmod{m_1} \\
-      &\ldots \\
+      &\vdots \\
     x &\equiv a_k \pmod{m_k}
 \end{align}$$
 
 (As above, assume that $m = m_1 m_2 \cdots m_k$ and $m_i$ are pairwise coprime).
 
+## Solution for Two Moduli
+
+Consider a system of two equations for coprime $m_1, m_2$:
+
+$$
+\begin{align}
+    a &\equiv a_1 \pmod{m_1} \\
+    a &\equiv a_2 \pmod{m_2} \\
+\end{align}
+$$
+
+We want to find a solution for $a \pmod{m_1 m_2}$. Using the [Extended Euclidean Algorithm](extended-euclid-algorithm.md) we can find Bézout coefficients $n_1, n_2$ such that
+
+$$n_1 m_1 + n_2 m_2 = 1$$
+
+Equivalently, $n_1 m_1 \equiv 1 \pmod{m_2}$ so $n_1 \equiv m_1^{-1} \pmod{m_2}$, and vice versa $n_2 \equiv m_2^{-1} \pmod{m_1}$. 
+
+Then a solution will be 
+
+$$a = a_1 n_2 m_2 + a_2 n_1 m_1$$
+
+We can easily verify $a = a_1 (1 - n_1 m_1) + a_2 n_1 m_1 \equiv a_1 \pmod{m_1}$ and vice versa.
+
+## Solution for General Case
+
+### Inductive Solution
+
+As $m_1 m_2$ is coprime to $m_3$, we can inductively repeatedly apply the solution for two moduli for any number of moduli. For example, combine $a \equiv b_2 \pmod{m_1 m_2}$ and $a \equiv a_3 \pmod{m_3}$ to get $a \equiv b_3 \pmod{m_1 m_2 m_3}$, etc.
+
+### Direct Construction
+
+A direct construction similar to Lagrange interpolation is possible. Let $M_i = \prod_{i \neq j} m_j$, the product of all moduli but $m_i$. Again with the Extended Euclidean algorithm we can find $N_i, n_i$ such that
+
+$$N_i M_i + n_i m_i = 1$$
+
+Then a solution to the system of congruences is
+
+$$a = \sum_{i=1}^k a_i N_i M_i$$
+
+Again as $N_i \equiv M_i^{-1} \pmod{m_i}$, the solution is equivalent to 
+
+$$a = \sum_{i=1}^k a_i M_i (M_i^{-1} \mod{m_i})$$
+
+Observe $M_i$ is a multiple of $m_j$ for $i \neq j$, and 
+
+$$a \equiv a_i N_i M_i \equiv a_i (1 - n_i m_i) \equiv a_i \pmod{m_i}$$
+
```

Comment:
Btw, I'm not sure if you saw this (as I posted it on a closed issue). https://github.com/cp-algorithms/cp-algorithms/pull/1013

I think we should add an implementation. E.g. one in C++.
And we don't need to worry about large numbers. In most contests, as far as I remember, M is not that big, and usually fits inside an 32 bit integer. So doing all computations is with 64 bit integers is fine.
E.g. a prime example is [Code Golf: Golf Gophers](https://codingcompetitions.withgoogle.com/codejam/round/0000000000051635/0000000000104f1a#problem). The solution requires the CRT to find the solution, but all computations can be done without any big integer library.

And if it's not fine for some harder problem, then the reader can himself use/implement a big integer library or switch to some other language.
It's not necessary for an educational website to provide copy-pasteable snippets for every single usecase.

This here is my personal implementation for the algorithm you explained (with some different variable names though).

```cpp
struct Congruence
{
    long long a, m, totient_m;
};

class CRT
{
public:
    void add_congruence(long long a, long long m, long long totient_m) {
        congruences.push_back({a, m, totient_m});
    }

    long long get_unique_solution() {
        long long M = 1;
        for (const auto& c : congruences) {
            M *= c.m;
        }

        long long solution = 0;
        for (const auto& c : congruences) {
            auto b = M / c.m;
            solution = (solution + c.a * b % M * power(b, c.totient_m - 1, c.m)) % M;
        }
        return solution;
    }

    std::vector<Congruence> congruences;
};
```